docs/guides/modules/test/pages/smarter-testing.adoc

```diff
@@ -41,565 +41,1571 @@ TIP: In codebases with sparse test coverage, Smarter Testing cannot accurately d
+----
+
+Yarn with Jest::
++
+[source,yaml]
+----
+# .circleci/test-suites.yml
+---
+name: ci tests
+discover: yarn --silent test --listTests
+run: JEST_JUNIT_OUTPUT_FILE="<< outputs.junit >>" yarn test --runInBand --reporters=jest-junit --bail << test.atoms >>
+analysis: |
+  yarn test --runInBand \
+            --coverage \
+            --coverageProvider=v8 \
+            --coverageReporters=lcovonly \
+            --coverage-directory="$(dirname << outputs.lcov >>)" \
+            --bail \
+            << test.atoms >> \
+            && cat "$(dirname << outputs.lcov >>)"/*.info > << outputs.lcov >>
+outputs:
+  junit: test-reports/tests.xml
+options:
+  test-impact-analysis: true
+  test-analysis-duration: 15
+----
+
+pytest::
++
+[source,yaml]
+----
+# .circleci/test-suites.yml
+---
+name: ci tests
+discover: find ./tests -type f -name 'test*.py'
+run: |
+  pytest --disable-pytest-warnings \
+         --no-header \
+         --quiet \
+         --tb=short \
+         --junit-xml="<< outputs.junit >>" \
+         << test.atoms >>
+analysis: |
+  pytest --disable-pytest-warnings \
+         --no-header \
+         --quiet \
+         --tb=short \
+         --cov \
+         --cov-report=lcov:<< outputs.lcov >> \
+         << test.atoms >>
+outputs:
+  junit: test-reports/tests.xml
+options:
+  test-impact-analysis: true
+  test-analysis-duration: 15
+----
+
+Go::
++
+[source,yaml]
+----
+# .circleci/test-suites.yml
+---
+name: ci tests
+discover: go list -f '{{ if or (len .TestGoFiles) (len .XTestGoFiles) }} {{ .ImportPath }} {{end}}' ./...
+run: go test -race -count=1 << test.atoms >>
+analysis: go test -coverprofile="<< outputs.go-coverage >>" -cover -coverpkg ./... << test.atoms >>
+outputs:
+  junit: test-reports/tests.xml
+options:
+  test-impact-analysis: true
+  test-analysis-duration: 15
+----
+
+Go with gotestsum::
++
+[source,yaml]
+----
+# .circleci/test-suites.yml
+---
+name: ci tests
+discover: go list -f '{{ if or (len .TestGoFiles) (len .XTestGoFiles) }} {{ .ImportPath }} {{end}}' ./...
+run: go tool gotestsum --junitfile="<< outputs.junit >>" -- -race -count=1 << test.atoms >>
+analysis: go tool gotestsum -- -coverprofile="<< outputs.go-coverage >>" -cover -coverpkg ./... << test.atoms >>
+outputs:
+  junit: test-reports/tests.xml
+options:
+  test-impact-analysis: true
+  test-analysis-duration: 15
+----
+====
+
+Run analysis, this will run until all the tests are analysed or for approximately the `test-analysis-duration` timeout you set in your test suite configuration. If the analysis command exits because it runs out of time, you can run the test suite again, the process will pick up from where it left off.
+[source,console]
+----
+$ circleci run testsuite "ci tests" --local --test-selection=none --test-analysis=impacted
+----
+
+Once the analysis is complete, modify a source file and then run the test-suite. Look for the section starting `Selecting tests...` (some of the output from the command below has been elided):
```

Comment:
Waiting for the full analysis can create a long feedback loop. Since the goal is just to check if this is set, we could suggest the user modify a test impacted by one of the tests shown in the discovery output. That way, the file will be selected while others are skipped. What do you think?